src/pages/SignUp/signUp.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from 'react';
+import UserNav from '../../components/userNav/userNav';
+import InputData from './InputData';
+import INPUT_LIST from './inputList';
+import './SignUp.scss';
+
+export class SignUp extends Component {
+  constructor() {
+    super();
+    this.state = {
+      id: '',
+      password: '',
+      pwCheck: '',
+      email: '',
+      userName: '',
+      birth: '',
+      gender: '',
+    };
+  }
+
+  submitUserInfoToSignUp = e => {
+    const { id, password, userName, email, birth, gender } = this.state;
+    fetch('http://10.58.5.138:8000/users/signup', {
+      method: 'POST',
+      body: JSON.stringify({
+        id: id,
+        name: userName,
+        password: password,
+        email: email,
+        yearOfBirth: birth,
+        gender: gender,
+      }),
+    })
+      .then(response => response.json())
+      .then(data => {
+        if (data.message === 'Id format is not valid') {
+          alert('유효하지않은 아이디입니다.');
+        } else if (data.message === 'Password_Validation_Error') {
+          alert('유효하지 않은 비밀번호입니다.');
+        } else if (data.message === 'Email format is not valid') {
+          alert('유효하지 않은 이메일입니다.');
+        } else if (data.message === 'ID_Exist_Error') {
+          alert('이미 등록된 아이디입니다.');
+        } else if (data.message === 'Email.Exist_Error') {
+          alert('이미 등록된 이메일입니다.');
+        } else if (data.message === 'SUCCESS') {
+          alert('회원가입 성공');
+          this.props.history.push('/Login');
+        }
+      });
+  };
+
+  handleInputs = e => {
+    const { name, value } = e.target;
+
+    this.setState({
+      [name]: value,
+    });
+  };
+
+  pushMan = () => {
+    this.setState({
+      gender: '1',
+    });
+  };
+
+  pushWoman = () => {
+    this.setState({
+      gender: '2',
+    });
+  };
+  render() {
+    const { pwCheck, password } = this.state;
+    const passwordConfirmed = pwCheck === password;
+
+    return (
+      <div className="totalContainer">
+        <UserNav />
+        <div className="signUpBody">
+          <section className="signUpSection">
+            {INPUT_LIST.map(inputList => {
+              return (
+                <InputData
+                  key={inputList.id}
+                  name={inputList.name}
+                  passwordConfirmed={passwordConfirmed}
+                  passwordCheck={this.passwordCheck}
+                  inputType={inputList.inputType}
+                  placeholder={inputList.placeholder}
+                  maxlength={inputList.maxLength}
+                  errormessage={inputList.errormessage}
+                  handleInputs={this.handleInputs}
+                />
```

Comment:
```suggestion
                <InputData
                  key={inputList.id}
                  inputList={inputList}
                  passwordConfirmed={passwordConfirmed}
                  passwordCheck={this.passwordCheck}
                  handleInputs={this.handleInputs}
                />
```
이런식으로 줄인다면 자식컴포넌트로 구조분해할당시 this.props.inputList로 정의해주시면 됩니다!

Reply:
리뷰 주신대로 반영했습니다!